routingpy/client_default.py

```diff
@@ -101,7 +37,7 @@ def __init__(
         retry_timeout=None,
         requests_kwargs=None,
```

Comment:
Let's remove this `requests_kwargs` entirely and change the base client to take a `**kwargs` as last argument. makes it more generic. Also remember to change all the "router" instances' `__init__`: in those we can call the argument `**client_kwargs` instead of `**kwargs` to make it more explicit.

I already made peace with releasing a major version update, so we can break existing stuff.